microservices/strapi-connector-ms/plugins/auth.js

```diff
@@ -0,0 +1,53 @@
+import got from 'got'
+import { createVerifier, TOKEN_ERROR_CODES } from 'fast-jwt'
+
+export async function asyncVerifyJWT (request, reply, done) {
+    const fastify = request.server
+
+    const authHeader = request.headers.authorization
+    if (authHeader) {
+        request.log.debug({ header: authHeader }, { userEndpoint: fastify.config.USER_ENDPOINT }, "asyncVerifyJWT")
+        try {
+            await got(
+                fastify.config.USER_ENDPOINT,
+                { headers: { "Authorization": authHeader }}
+            )
+            // done()
+        } catch (err) {
+            request.log.warn(err.response.body)
+            return reply.code(err.response.statusCode).send(err.response.body)
+        }
+    } else {
+        request.log.warn("asyncVerifyJWT: missing_header")
+        return reply.code(400).send({ "error": "missing_header", "error_description": "Authorization header not set" })
+    }
+}
+
+export function pubKeyVerifyJWT(request, reply, done) {
+    const fastify = request.server
+
+    const pemHeader = "-----BEGIN CERTIFICATE-----\n"
+    const pemFooter = "\n-----END CERTIFICATE-----"
+    const pubKeyString = fastify.config.JWT_PUB_KEY
+
+    const pubKey = pemHeader.concat(pubKeyString, pemFooter)
+    const authHeader = request.headers.authorization
+    if (authHeader && authHeader.split(' ')[0] == 'Bearer') {
```

Comment:
I suggest to change the check considering case insensitivity:
```
authHeader.split(' ')[0].toLowerCase() === 'bearer'
```
HTTP headers are case insensitive and some clients send them lowercase. Moreover HTTP/2 encodes HTTP headers as lowercase by design.

Reply:
Good call, didn't know about HTTP/2 encoding to lowercase. Fixed: https://github.com/entando/entando-strapi-connector/pull/6/commits/8b80c24330fbe670d12691e3cc2719dce472f6c2